apps/files_versions/lib/Listener/LegacyRollbackListener.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\Files_Versions\Listener;
+
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCA\Files_Versions\Events\VersionRestoredEvent;
+
+/**
+ * This listener is designed to be compatible with third-party code
+ * that can still use a hook. This listener will be removed in
+ * the next version and the rollback hook will stop working.
+ *
```

Comment:
```suggestion
 *
 * @deprecated 32.0.0
```

Makes it easier to find it later